public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+
+:marked
+  `FormBuilder.group` is a factory method that creates a `FormGroup`. &nbsp;
+  `FormBuilder.group` takes an object whose keys and values are `FormControl` names and their definitions. 
+  In this example, the `name` control is defined by its initial data value, an empty string.
+
+  Defining a group of controls in a single object makes for a compact, readable style.
+  It beats writing an equivalent series of `new FormControl(...)` statements.
+
+  The form looks great. But does it work? 
+  When the user enters a name, where does the value go? 
+  
+  The value goes into the **_form model_** that backs the group's `FormControls`.
+  To see the form model, add the following line after the 
+  closing `form` tag in the `hero-detail.component.html`:
+
++makeExample('reactive-forms/ts/app/hero-detail-2.component.html', 'form-value-json','app/hero-detail.component.html')(format=".")
+
+:marked
+  The `heroForm.value` returns the _form model_.
+  Piping it through the `JsonPipe` renders the model as JSON in the browser:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/json-output.png" width="400px" alt="JSON output")
+
+:marked
+  The initial `name` property value is the empty string. 
+  Type into the _name_ input box and watch the keystokes appear in the JSON.
+
+  ### More FormControls
+  A hero has more than a name. A hero has an address too. 
+  Add some address `FormControls` to the `heroForm` as follows.
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v3','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  Then add corresponding markup in `hero-detail.component.html` 
+  within the `form` element.
+
++makeExample('reactive-forms/ts/app/hero-detail-3.component.html', '','app/hero-detail.component.html')(format=".")
+
+.alert.is-helpful
+  :marked
+    *Reminder*: Ignore the many mentions of `form-group` and `form-control`in this markup.
+    Those are _bootstrap_ CSS classes that Angular itself ignores.
+    Pay attention to the `formGroupName` and `formControlName` attributes.
+    They are the Angular directives that bind the HTML controls to the 
+    Angular `FormGroup` and `FormControl` properties in the component class.
+
+.l-main-section
+a#grouping
+:marked
+  ### Group FormControls
+
+  This form is getting big and unwieldy.
+  You can make it more maintainable by grouping some of the related `FormControls` into another `FormGroup`.
+  The `street`, `city`, `state`, and `zip` are properties that would make a good _address_ `FormGroup`. 
+  
+  You used the `FormBuilder` to create one `FormGroup` in this component called `heroForm`.
+  Let that be the parent `FormGroup`.
+  Use `FormBuilder` again to create a child `FormGroup` that encapsulates the address controls;
+  assign the result to a new `address` property of the parent `FormGroup`.
```

Comment:
We are so close to done. Unless there are strong objections, lets stick with what @kapunahelewong has here.